### DISCUSSION TITLE
framebuffer support ?

### DISCUSSION THREAD

**Discussion by grondilu** [kovidgoyal/kitty#6098] [Ideas]
I think opengl can run on the [framebuffer](https://fr.wikipedia.org/wiki/Framebuffer_Linux) (via say GLFBDev as mentioned in [this](https://archive.fosdem.org/2020/schedule/event/fbdev/attachments/slides/3595/export/events/attachments/fbdev/slides/3595/fosdem_2020_nicolas_caramelli_linux_framebuffer.pdf) for instance).

So, framebuffer support might be doable for kitty?  Wouldn't that be cool?

**Comment by kovidgoyal** [comment]
There's far more to framebuffer than just OpenGL. One has to handle
input events as well. You would need to write a GLFW backend for the
framebuffer. And that would be difficult since GLFW assumes you can
create multiple windows, while a framebuffer has no window manager.

But feel free to try.

**Comment by defdefred** [comment]
@grondilu You may be interested by https://github.com/borisfaure/terminology for framebuffer support.
I just test it with Debian testing, the package is compiled with framebuffer support.
